tube/spark/indexers/aggregation/lambdas.py

```diff
@@ -0,0 +1,64 @@
+def merge_arrays(x, y):
+    if x is None and y is None:
+        return []
+    elif x is None and y is not None:
+        return y
+    elif x is not None and y is None:
+        return x
+    else:
+        return list(set(x) | set(y))
+
+
+def get_aggregation_func_by_name(func_name, is_merging=False):
+    if func_name == 'count':
+        if is_merging:
+            return lambda x, y: x + y
+        return lambda x, y: x + 1
+    if func_name == 'sum':
+        return lambda x, y: x + y
+    if func_name == 'set':
+        return lambda x, y: merge_arrays(x, y)
+
+
+def get_single_frame_zero_by_func(func_name, output_name):
+    if func_name == 'set':
+        return (func_name, output_name, [])
+    if func_name == 'count' or func_name == 'sum':
+        return (func_name, output_name, 0)
+    return (func_name, output_name, '')
+
+
+def get_single_frame_value(func_name, value):
+    if func_name == 'set':
+        return [value] if type(value) is not list else value
+    if func_name == 'count' or func_name == 'sum':
+        return value
+    return ''
+
+
+def get_frame_zero(reducers):
+    return tuple([get_single_frame_zero_by_func(rd.fn, rd.prop.name) for rd in reducers if not rd.done])
+
+
+def get_normal_frame(reducers):
+    return lambda x: tuple([(rd.fn, rd.prop.name, get_single_frame_value(rd.fn, x.get(rd.prop.name)))
+                            for rd in reducers if not rd.done])
+
+
+def seq_aggregate_with_reducer(x, y):
```

Comment:
Why have two different functions, instead of a parameter?

Can you introduce some "naming" for indices (0, 1, 2)? So it's clear that 0 is the function name and 2 is the value.

Reply:
https://tdhopper.com/blog/pysparks-aggregatebykey-method/
there are two steps in map-reduce, when they send the record to each other and then merge everything. `seq_aggregate_with_reducer` is used for the first step, `merge_` is used for the second one.
(0, 1, 2) -> ('func_name', 'field_name', 'value') this is convention tuple in tube to let the worker know what function should be perform. Will add in a comment in the code